homeassistant/components/mqtt/remote.py

```diff
@@ -0,0 +1,238 @@
+"""Support for MQTT remotes."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import mqtt
+from homeassistant.components.remote import RemoteDevice
+from homeassistant.const import (
+    CONF_COMMAND,
+    CONF_DEVICE,
+    CONF_ICON,
+    CONF_NAME,
+    CONF_OPTIMISTIC,
+    CONF_PAYLOAD_OFF,
+    CONF_PAYLOAD_ON,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from . import (
+    CONF_COMMAND_TOPIC,
+    CONF_QOS,
+    CONF_RETAIN,
+    CONF_STATE_TOPIC,
+    CONF_UNIQUE_ID,
+    MqttAttributes,
+    MqttAvailability,
+    MqttEntityDeviceInfo,
+    subscription,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "MQTT Remote"
+DEFAULT_OPTIMISTIC = False
+DEFAULT_PAYLOAD_ON = "ON"
+DEFAULT_PAYLOAD_OFF = "OFF"
+
+CONF_COMMANDS = "commands"
+
+COMMAND_SCHEMA = vol.Schema(
+    {vol.Required(CONF_COMMAND): vol.All(str, vol.Length(min=1))}
+)
+
+PLATFORM_SCHEMA = (
+    mqtt.MQTT_RW_PLATFORM_SCHEMA.extend(
+        {
+            vol.Optional(CONF_DEVICE): mqtt.MQTT_ENTITY_DEVICE_INFO_SCHEMA,
+            vol.Optional(CONF_ICON): cv.icon,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+            vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
+            vol.Optional(CONF_PAYLOAD_OFF, default=DEFAULT_PAYLOAD_OFF): cv.string,
+            vol.Optional(CONF_PAYLOAD_ON, default=DEFAULT_PAYLOAD_ON): cv.string,
+            vol.Optional(CONF_STATE_TOPIC): mqtt.valid_subscribe_topic,
+            vol.Optional(CONF_COMMANDS, default={}): cv.schema_with_slug_keys(
+                COMMAND_SCHEMA
```

Comment:
Why is `COMMAND_SCHEMA` a dictionary if it only holds one value?

Reply:
CONF_COMMANDS can hold multiple commands. For example:

    commands:
      vol_up:
        command: "volume_up"
      vol_down:
        command: "volume_down"